### PR TITLE
feature: add module with Hashicorp Vault

### PR DESCRIPTION
**Pull request by wallhided**
sak issue #189 

Known problems: 
Execute manual command for initiate vault (Anyway I can use local_exec)

### CODE REVIEW

File: hashicorp-vault/main.tf [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  tags = var.tags
+}
+
+resource "aws_iam_access_key" "this" {
+  count = var.s3_storage ? 1 : 0
+  user  = aws_iam_user.this.0.name
+}
+
+
+resource "helm_release" "this" {
```
[the remark this answers is not in the document]
Comment (wallhided) [reply hashicorp-vault/main.tf:128]: added below

File: hashicorp-vault/main.tf [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        Effect   = "Allow"
+        Resource = "*"
+      },
+    ]
+  })
+}
+
+resource "aws_iam_policy_attachment" "this" {
+  count      = var.s3_storage ? 1 : 0
+  name       = "test-attachment"
```
Comment (akastav) [inline hashicorp-vault/main.tf:78]: It would be better to use the module name + cluster name to ensure uniqueness

File: hashicorp-vault/main.tf [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "s3:*"
+        ]
+        Effect   = "Allow"
+        Resource = "*"
```
Comment (akastav) [inline hashicorp-vault/main.tf:70]: why do I need access to all s3 buckets? Too wide a set of rights, you need to restrict access only to your s3 bucket

File: hashicorp-vault/main.tf [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+      name  = set.key
+      value = set.value
+    }
+  }
+}
+
+locals {
+  name           = var.chart_name
+  repository     = "https://helm.releases.hashicorp.com"
+  chart          = "vault"
```
Comment (akastav) [inline hashicorp-vault/main.tf:149]: You can use var.chart_name here too

File: hashicorp-vault/main.tf [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+storage "file" {
+  path = "/vault/data"
+}
+EOT
+    } : {},
+    {
+  })
+
+  vault_conf = <<EOT
+ui = true
```
Comment (akastav) [inline hashicorp-vault/main.tf:217]: If we set UI = true, we need to add some properties for ingress URL and annotations like a cert-manager

File: hashicorp-vault/variables.tf [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+variable "chart_name" {
+  type        = string
+  default     = "hashicorp-vault"
+  description = "A chart name for vault installation"
+}
+
+variable "chart_namespace" {
+  type        = string
+  default     = "default"
```
Comment (akastav) [inline hashicorp-vault/variables.tf:20]: default namespace allow access to anybody who has access to the cluster, much better used kube-system namespace

File: hashicorp-vault/main.tf (hunk 4 above)
Comment (wallhided) [reply hashicorp-vault/main.tf:149]: I think it is not a good idea, cuz name "vault" related to chart name in repo.

File: hashicorp-vault/main.tf (hunk 5 above)
Comment (wallhided) [reply hashicorp-vault/main.tf:217]: ok, I'll remove this. Seems like it's redundant.

**Review by RustamGimadiev — APPROVED** [review]

**Review by akastav — APPROVED** [review]

**Pull request merged**